bin/desi_qso_qn_afterburner

```diff
@@ -341,14 +341,20 @@ def save_dataframe_to_fits(dataframe, filename, DESI_TARGET, clobber=True):
     data['Z_Hbeta'] = np.array([dataframe['Z_LINES'][i][4] for i in range(dataframe.shape[0])])
     data['Z_Halpha'] = np.array([dataframe['Z_LINES'][i][5] for i in range(dataframe.shape[0])])
 
-    fits = fitsio.FITS(filename, 'rw', clobber=clobber)
+    # Save file in temporary file to track when timeout error appears during the writing
+    tmpfile = get_tempfilename(filename)
+    fits = fitsio.FITS(tmpfile, 'rw', clobber=clobber)
     fits.write(data, extname='QN_RR')
     if clobber:
         log.info(f'OVERWRITE the file : {filename}')
     else:
         log.info(f'EXPAND the file : {filename}')
     fits.close()
 
+    # Rename temporary file to the
+    os.rename(tmpfile, filename)
```

Comment:
See above comment about appending to fits files when using a temporary file; this section will need to be updated too.